### PR TITLE
VPN Landing page updates (Fixes #9950)

### LINKED ISSUE #9950: VPN Landing page: updates & localization
## Description

- Update the VPN landing page design based on of qualitative testing.
- Open up the page for L10n.

---

#### Success Criteria
- [ ] Updated VPN landing page is exposed in Pontoon for L10n.

### PR DESCRIPTION
**Pull request by alexgibson**
## Description
- Update VPN landing page design and copy.
- Expose strings in Pontoon for L10n.
- Adds a multi-variant headline experiment for `en-US / de / fr` locales using Traffic Cop (behind the switch `vpn-landing-page-heading-experiment`).

URLs:
- http://localhost:8000/en-US/products/vpn/
- http://localhost:8000/en-US/products/vpn/invite/

Experiment URLs:
- http://localhost:8000/en-US/products/vpn/?entrypoint_experiment=vpn-landing-page-heading&entrypoint_variation=a (control)
- http://localhost:8000/en-US/products/vpn/?entrypoint_experiment=vpn-landing-page-heading&entrypoint_variation=b
- http://localhost:8000/en-US/products/vpn/?entrypoint_experiment=vpn-landing-page-heading&entrypoint_variation=c
- http://localhost:8000/en-US/products/vpn/?entrypoint_experiment=vpn-landing-page-heading&entrypoint_variation=d

## Issue / Bugzilla link
#9950

## Testing
- https://www-demo1.allizom.org/en-US/products/vpn/ (VPN available)
- https://www-demo1.allizom.org/en-US/products/vpn/?geo=de (VPN not available)
- https://www-demo1.allizom.org/en-US/products/vpn/invite/

**Note:** to test the traffic cop headline experiment you'll need to use a browser with **Do Not Track disbaled**.

**Things worth checking:**

- [ ] Does page responsiveness look OK?
- [ ] All strings are localized?
- [ ] Does conditional logic for VPN availability work as expected and make sense?
- [ ] Does the Traffic Cop config look OK? Does the experiment trigger as expected?

### CODE REVIEW

File: l10n/en/products/vpn/shared.ftl
Comment (alexgibson) [inline l10n/en/products/vpn/shared.ftl:1]: @maxxcrawford this is what I was thinking for shared strings, let me know your thoughts! 

Currently this includes things like:

- CTA button text.
- Available countries where VPN is offered.
- Supported platform info.
- Pricing info.
- Product features.
- Privacy and copyright text.

Strings here will almost certainly change in the next week or so, but hopefully we can align on what will be used here between pages.

Also, if there are any other strings you think might be useful here for the SEO pages then let me know! Likewise, if there are common elements or sections of the landing page that could be shared on other SEO pages, then I can look to make those includes at the template level.

File: l10n/en/products/vpn/shared.ftl
Comment (maxxcrawford) [inline l10n/en/products/vpn/shared.ftl:23]: Do we need to localize the latest list of countries too? _(Writing out each country name, comma seperated?)_

File: l10n/en/products/vpn/shared.ftl
Comment (maxxcrawford) [reply l10n/en/products/vpn/shared.ftl:1]: This looks/works great. I'm not sure about shared components yet. I keep thinking the pricing box and cta may be one component.

File: l10n/en/products/vpn/shared.ftl
Comment (alexgibson) [reply l10n/en/products/vpn/shared.ftl:23]: We have that too in the string `vpn-available-countries` 👍

Comment (maxxcrawford) [reply l10n/en/products/vpn/shared.ftl:23]: 🙈

File: l10n/en/products/vpn/shared.ftl
Comment (maxxcrawford) [reply l10n/en/products/vpn/shared.ftl:1]: Okay – I checked out the copy doc for the other pages and found only a few strings that were repeated (without some variation): 

- What you’ll get with Mozilla VPN
- Protection for up to 5 devices
- Access to servers in 30+ countries
- Device-level encryption
- No bandwidth restrictions
- No online activity logs now or ever
- Connect up to 5 devices
- 30-day money-back guarantee

Comment (maxxcrawford) [reply l10n/en/products/vpn/shared.ftl:1]: Discalimer: Some of these lines are not finalized, but they are used in each section. (Up to five in use on each page)

Comment (alexgibson) [reply l10n/en/products/vpn/shared.ftl:1]: @maxxcrawford most of the strings you mention here in relation to VPN features are almost (but not quite) the same as the landing page strings. Perhaps this is something we could try and standardize on once we're closer to final copy 👍

File: l10n/en/products/vpn/landing.ftl
Comment (peiying2) [inline l10n/en/products/vpn/landing.ftl:60]: Brand tagging in `WireGuard’s lightweight code ...`

File: l10n/en/products/vpn/landing.ftl
Comment (peiying2) [inline l10n/en/products/vpn/landing.ftl:108]: Add a comma after `By clicking "Joining the waitlist"`

File: l10n/en/products/vpn/shared.ftl
Comment (peiying2) [inline l10n/en/products/vpn/shared.ftl:60]: I assume `$amount (string)` is considered a string instead of a number is because of the currency associated with the number? And the currency can be automatically generated, depending the currency.

File: l10n/en/products/vpn/shared.ftl
Comment (peiying2) [inline l10n/en/products/vpn/shared.ftl:50]: Add a `be` in `... only be shown ...`

File: l10n/en/products/vpn/shared.ftl
Comment (peiying2) [inline l10n/en/products/vpn/shared.ftl:15]: What is followed after this? I don't see a lot of the strings in the shared file.

File: l10n/en/brands.ftl
Comment (peiying2) [inline l10n/en/brands.ftl:143]: This is an entry added by maxx in his recent PR

File: l10n/en/products/vpn/shared.ftl
Comment (alexgibson) [reply l10n/en/products/vpn/shared.ftl:60]: Yes we'll provide both the amount and the currency. I'll make this clearer in the comment

File: l10n/en/products/vpn/shared.ftl
Comment (alexgibson) [reply l10n/en/products/vpn/shared.ftl:15]: It will be a lockup of press logos (not in the page just yet until we get approvals). I'll add a comment here to provide context.

**Comment by alexgibson** [comment]
Thanks @peiying2 comments addressed 💯

File: bedrock/products/templates/products/vpn/includes/featured-press.html
Comment (alexgibson) [inline bedrock/products/templates/products/vpn/includes/featured-press.html:1]: Note for reviewer: this file isn't included yet (we're waiting on budget/approvals for using the logos). It's OK to merge this as-is and I'l include it in the [next update](https://github.com/mozilla/bedrock/issues/9948).

File: bedrock/products/templates/products/vpn/includes/pricing-fixed.html
Comment (alexgibson) [inline bedrock/products/templates/products/vpn/includes/pricing-fixed.html:1]: Note for reviewer: this file also isn't included in the page yet. I'm adding it here as placeholder so we can start lcoalizing the strings needed for https://github.com/mozilla/bedrock/issues/9948

File: bedrock/products/templates/products/vpn/includes/pricing-variable.html
Comment (alexgibson) [inline bedrock/products/templates/products/vpn/includes/pricing-variable.html:1]: Note for reviewer: this file also isn't included in the page yet. I'm adding it here as placeholder so we can start lcoalizing the strings needed for https://github.com/mozilla/bedrock/issues/9948

The styling and layout is still WIP here.

File: bedrock/products/views.py
Comment (alexgibson) [inline bedrock/products/views.py:23]: Note for reviewer: these variable prices are just placeholder for now and not yet used in the page.

**Comment by alexgibson** [comment]
@maxxcrawford r?

Let's hold off merging until Monday, but this is now good for code review. If you want to chat about anything here, let me know and I'd be happy to sync. Thanks!

File: l10n/en/products/vpn/shared.ftl
Comment (maxxcrawford) [inline l10n/en/products/vpn/shared.ftl:36]: nit/copy: Should there be a `+` after each variable?

File: l10n/en/products/vpn/shared.ftl
Comment (maxxcrawford) [inline l10n/en/products/vpn/shared.ftl:50]: Does this also need a `®` placed anywhere?

File: media/css/products/vpn/components/block.scss
Comment (maxxcrawford) [inline media/css/products/vpn/components/block.scss:21]: Should this be a protocol token?

File: media/css/products/vpn/components/block.scss
Comment (maxxcrawford) [inline media/css/products/vpn/components/block.scss:53]: Should this be a protocol token?

File: media/css/products/vpn/components/block.scss
Comment (maxxcrawford) [inline media/css/products/vpn/components/block.scss:123]: Should this be a protocol token? `$screen-sm`

File: media/css/products/vpn/components/block.scss
Comment (maxxcrawford) [inline media/css/products/vpn/components/block.scss:150]: Should this be a protocol token?

File: media/css/products/vpn/components/hero.scss
Comment (maxxcrawford) [inline media/css/products/vpn/components/hero.scss:25]: Should this be a protocol token? `$screen-sm`

File: media/css/products/vpn/components/hero.scss
Comment (maxxcrawford) [inline media/css/products/vpn/components/hero.scss:61]: Should this be a protocol token?

File: media/css/products/vpn/components/hero.scss
Comment (maxxcrawford) [inline media/css/products/vpn/components/hero.scss:64]: Should this be a protocol token?

File: media/css/products/vpn/components/hero.scss
Comment (maxxcrawford) [inline media/css/products/vpn/components/hero.scss:86]: Should this be a protocol token? `$screen-sm`

File: media/css/products/vpn/components/hero.scss
Comment (maxxcrawford) [inline media/css/products/vpn/components/hero.scss:97]: Should this be a protocol token? `$screen-sm`

File: media/js/products/vpn/heading-experiment.js
Comment (maxxcrawford) [inline media/js/products/vpn/heading-experiment.js:41]: Comment: I love how easy TrafficCop is. 🙌

File: tests/functional/products/vpn/test_landing.py
Comment (maxxcrawford) [inline tests/functional/products/vpn/test_landing.py:13]: Naming convention: Should this be `get` instead of `try`?

**Review by maxxcrawford — CHANGES_REQUESTED** [review]
Outside of the SCSS comments _(swaping to tokens were applicable)_, I don't have anything blocking. Even the token swap is near non-blocking so if this needs to ship, I can revise.

Thanks!

File: bedrock/products/templates/products/vpn/includes/footer-legal.html
Comment (alexgibson) [reply bedrock/products/templates/products/vpn/includes/footer-legal.html:6]: Filed a separate issue for this: https://github.com/mozilla/bedrock/issues/10025

File: bedrock/products/templates/products/vpn/includes/footer-subscribe.html
Comment (alexgibson) [reply bedrock/products/templates/products/vpn/includes/footer-subscribe.html:17]: This is used to register the GA conversion goal, so should always be in English 👍

File: bedrock/products/templates/products/vpn/includes/footer-subscribe.html
Comment (alexgibson) [reply bedrock/products/templates/products/vpn/includes/footer-subscribe.html:26]: Yes, keeping

File: bedrock/products/templates/products/vpn/invite.html
Comment (alexgibson) [reply bedrock/products/templates/products/vpn/invite.html:54]: For locales that specify a country (e.g. `en-US`, `es-ES`), the form will preselect a country. We don't do any quessing on people's country for non-specific locale codes though, such as `de` etc.

File: bedrock/products/templates/products/vpn/landing.html
Comment (alexgibson) [reply bedrock/products/templates/products/vpn/landing.html:25]: en-US only is the requirement so keeping things simple

File: bedrock/products/templates/products/vpn/variants/heading-b.html
Comment (alexgibson) [reply bedrock/products/templates/products/vpn/variants/heading-b.html:14]: These strings are just for an experiment so hard coding is fine for now. We'll localize the winner once the experiment is complete.

File: l10n/en/products/vpn/shared.ftl
Comment (alexgibson) [reply l10n/en/products/vpn/shared.ftl:36]: The copy says "more than..." and "in over..." so omiting the `+` is OK here I think.

File: media/css/products/vpn/common.scss
Comment (alexgibson) [reply media/css/products/vpn/common.scss:72]: These values are set to match the specific image dimensions, so I think a hard-coded value makes sense here.

File: media/css/products/vpn/components/block.scss
Comment (alexgibson) [reply media/css/products/vpn/components/block.scss:21]: Hmm, I'm not sure we have any specic tokens for border sizing/styling in Protocol?

File: media/css/products/vpn/components/block.scss
Comment (alexgibson) [reply media/css/products/vpn/components/block.scss:123]: This value is matching the specific image size, so I'm not sure a screen-sizing token makes sense

File: media/css/products/vpn/components/hero.scss
Comment (alexgibson) [reply media/css/products/vpn/components/hero.scss:34]: These are the specific sizes of the wordmark / logo, so they need to match.

File: tests/functional/products/vpn/test_landing.py
Comment (alexgibson) [reply tests/functional/products/vpn/test_landing.py:13]: Good suggestion!

File: media/css/products/vpn/components/block.scss
Comment (maxxcrawford) [reply media/css/products/vpn/components/block.scss:21]: Ahh – I was thinking about `$border-radius-xs`… but that's for `border-radius`. Please disregard!

**Comment by alexgibson** [comment]
Thanks for the review @maxxcrawford! I think most of the token suggestions are in places where the CSS is referencing specific image dimensions, so hard-coded values are probably less prone to change unexpectedly.

I added a commit to update the test locator names. Appreciate the careful reivew! 💯

**Comment by alexgibson** [comment]
Successful test run: https://gitlab.com/mozmeao/www-config/-/pipelines/273414123

**Review by maxxcrawford — APPROVED** [review]
🍃

**Pull request merged**
